web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -302,16 +308,15 @@ export function EnrollEksCluster(props: AgentStepProps) {
         emitErrorEvent(
           'unknown error: no results came back from enrolling the EKS cluster.'
         );
-      } else if (
-        result.error &&
-        !result.error.includes(
-          'teleport-kube-agent is already installed on the cluster'
-        )
-      ) {
-        setEnrollmentState({
+      } else if (result.error) {
+        const errorState: EKSClusterEnrollmentState = {
           status: 'error',
           error: `Cluster enrollment error: ${result.error}`,
-        });
+        };
+        if (result.error.includes('teleport-kube-agent is already installed')) {
```

Comment:
```suggestion
        if (result.error.includes('teleport-kube-agent is already installed on the cluster')) {
```
We have to be stricter here because this string appears in two scenarios:
- agent is already installed
- could not determine if agent is already installed

https://github.com/gravitational/teleport/blob/3aa6f3c5cf71b81e61f4dd1653913e848e084981/lib/integrations/awsoidc/eks_enroll_clusters.go#L440-L445

Reply:
thanks for the catch